### PR TITLE
[AMDGPU] Split wide integer dpp8 intrinsic calls

### PR DESCRIPTION
**Pull request by rampitec**
The int_amdgcn_mov_dpp8 is declared with llvm_anyint_ty, but we can only select i32. To allow a corresponding builtin to be overloaded the same way as int_amdgcn_mov_dpp we need it to be able to split unsupported i64 values.

### CODE REVIEW

**Comment by rampitec** [comment]
Ping

File: llvm/lib/Target/AMDGPU/AMDGPUCodeGenPrepare.cpp [hunk 1]
```diff
@@ -2257,6 +2260,38 @@ bool AMDGPUCodeGenPrepareImpl::visitSqrt(IntrinsicInst &Sqrt) {
   return true;
 }
 
+// Split unsupported wide integer calls.
+bool AMDGPUCodeGenPrepareImpl::visitMovDppIntrinsic(IntrinsicInst &I) {
```
Comment (arsenm) [inline llvm/lib/Target/AMDGPU/AMDGPUCodeGenPrepare.cpp:2264]: This is an optimization pass and cannot be relied on for lowering. This should be handled in codegen proper like the other dpp intrinsics

Comment (rampitec) [reply llvm/lib/Target/AMDGPU/AMDGPUCodeGenPrepare.cpp:2264]: Didn't notice it does not happen at -O0. It looks like there is no place for it at -O0. The problem with 'other dpp intrinsics' is that it expands pseudo MI V_MOV_DPP_B64 only, past ISel, and does not handle any other types but i64. It almost calls for another pass.

Comment (arsenm) [reply llvm/lib/Target/AMDGPU/AMDGPUCodeGenPrepare.cpp:2264]: This is purely lowering. We already have the code to split arbitrary readfirstlane etc. intrinsics, this is no different (e.g. see 5feb32ba929f9e517c530217cabb09d1d734a763)

Comment (rampitec) [reply llvm/lib/Target/AMDGPU/AMDGPUCodeGenPrepare.cpp:2264]: Thanks. Dropping it in favor of https://github.com/llvm/llvm-project/pull/114296.

**Pull request closed without merging**